### PR TITLE
Graphql playground autodetects queries

### PR DESCRIPTION
**Pull request by zetlen**
_This PR is a cherry-pick from #1198 to make that PR more digestible._

It is number 4 of 6 in the chain of cherry-picked PRs. It needs to be merged _after_ these:

1. #1215
2. #1216 
3. #1217 

It needs to be merged before the next 2:

5. #1224
6. #1198

## Description
Instead of manually supplying query directories to search before starting up the GraphQL playground, this PR hooks into the Webpack compilation to find `.graphql` files and preloads the playground with them.

This makes Venia more portable, since not every Venia-consuming app needs a `queries` directory,

## Related Issue

Factoring this change out of #1198. 

## Verification Steps

1. Add a `.graphql` file anywhere in the project--in a component directory, for instance.
1. Import that file into a component used in the app.
1. Run the dev server (`yarn build && yarn watch:venia`).
1. Open the GraphQL playground to see your `.graphql` file in the preloaded tabs.

## How Have YOU Tested this?
Run devserver for UAT.

## Proposed Labels for Change Type/Package

- [ ] major (e.g x.0.0 - a breaking change)
- [x] minor (e.g 0.x.0 - a backwards compatible addition)
- [ ] patch (e.g 0.0.x - a bug fix)

## Checklist:

- [x] I have updated the documentation accordingly, if necessary.
- [x] I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/pwa-buildpack/src/WebpackTools/PWADevServer.js [hunk 1]
```diff
@@ -179,47 +178,48 @@ be configured to have the same effect as 'id'.
...
-                    return { dir, files };
-                })
-            );
+            const oldBefore = devServerConfig.before;
+            devServerConfig.before = (app, server) => {
+                let middleware;
+                let tabs;
+                oldBefore(app, server);
+                // this middleware has a bad habit of calling next() when it
+                // should not, so let's give it a noop next()
```
Comment (supernova-at) [inline packages/pwa-buildpack/src/WebpackTools/PWADevServer.js]: Let's move this comment down to line 221 where it applies

File: packages/pwa-buildpack/src/WebpackTools/PWADevServer.js [hunk 2]
```diff
@@ -179,47 +178,48 @@ be configured to have the same effect as 'id'.
...
+                    }
 
-            const queryDirContents = await Promise.all(
-                queryDirListings.map(({ dir, files }) =>
-                    Promise.all(
-                        files.map(async queryFile => {
-                            const fileAbsPath = resolve(dir, queryFile);
-                            const query = await readFile(fileAbsPath, 'utf8');
-                            const name = relative(process.cwd(), fileAbsPath);
+                    tabs = await Promise.all(
```
Comment (supernova-at) [inline packages/pwa-buildpack/src/WebpackTools/PWADevServer.js]: Do we have to `await` here?  This is the last statement of this function and the `await` on L215 will take care of waiting for the `Promise.all` to finish, right?

**Comment by zetlen** [comment]
@supernova-at Thanks for the PR feedback; I've addressed it!

**Comment by supernova-at** [comment]
@zetlen please see above, Danger is complaining about a failed unit test.

**Comment by dpatil-magento** [comment]
QA : UAT  and verification steps pass.

**Pull request merged**